libs/coin-modules/coin-multiversx/src/signOperation.ts

```diff
@@ -2,85 +2,93 @@ import { Observable } from "rxjs";
 import { FeeNotLoaded } from "@ledgerhq/errors";
 import { AccountBridge } from "@ledgerhq/types-live";
 import { buildOptimisticOperation } from "./buildOptimisticOperation";
-import { buildTransactionToSign } from "./buildTransaction";
+import { buildTransactionToSign, provideESDTInfo } from "./buildTransaction";
 import type { ElrondAccount, Transaction } from "./types";
-import { decodeTokenAccountId } from "../../account";
-import { withDevice } from "../../hw/deviceAccess";
+import { decodeTokenAccountId } from "@ledgerhq/coin-framework/account";
 import { extractTokenId } from "./logic";
 import { CHAIN_ID } from "./constants";
-import Elrond from "./hw-app-elrond";
+import { ElrondSigner } from "./signer";
+import { SignerContext } from "@ledgerhq/coin-framework/signer";
 
 /**
  * Sign Transaction with Ledger hardware
  */
-export const signOperation: AccountBridge<Transaction, ElrondAccount>["signOperation"] = ({
-  account,
-  deviceId,
-  transaction,
-}) =>
-  withDevice(deviceId)(
-    transport =>
-      new Observable(o => {
-        async function main() {
-          if (!transaction.fees) {
-            throw new FeeNotLoaded();
-          }
-          // Collect data for an ESDT transfer
-          const { subAccounts } = account;
-          const { subAccountId } = transaction;
-          const tokenAccount = !subAccountId
-            ? null
-            : subAccounts && subAccounts.find(ta => ta.id === subAccountId);
-
-          const elrond = new Elrond(transport);
-          await elrond.setAddress(account.freshAddressPath);
-
-          if (tokenAccount) {
-            const { token } = decodeTokenAccountId(tokenAccount.id);
-            if (token == null) {
-              throw new Error("Invalid token");
-            }
-
-            await elrond.provideESDTInfo(
-              token.ticker,
-              extractTokenId(token.id),
-              token.units[0].magnitude,
-              CHAIN_ID,
-              token.ledgerSignature,
-            );
+export const buildSignOperation =
+  (
+    signerContext: SignerContext<ElrondSigner>,
+  ): AccountBridge<Transaction, ElrondAccount>["signOperation"] =>
+  ({ account, transaction, deviceId }) =>
+    new Observable(o => {
+      let cancelled = false;
+
+      async function main() {
+        if (!transaction.fees) {
+          throw new FeeNotLoaded();
+        }
+        // Collect data for an ESDT transfer
+        const { subAccounts } = account;
+        const { subAccountId } = transaction;
+        const tokenAccount = !subAccountId
+          ? null
+          : subAccounts && subAccounts.find(ta => ta.id === subAccountId);
+
+        if (tokenAccount) {
+          const { token } = decodeTokenAccountId(tokenAccount.id);
+          if (token == null) {
+            throw new Error("Invalid token");
           }
 
-          const unsignedTx: string = await buildTransactionToSign(account, transaction);
+          await provideESDTInfo(
```

Comment:
We need to communicate with device for this operation.
On ledger device we need to enable a different flow for ESDT transfers.